src/commandsAndMenu.tsx

```diff
@@ -1439,3 +1402,73 @@ namespace Private {
   }
 }
 /* eslint-enable no-inner-declarations */
+
+export const gitCloneCommandPlugin: JupyterFrontEndPlugin<void> = {
```

Comment:
Would you mind moving this in its own file?

This will help readability of the code.

Reply:
I did not move it out because there was a reference to a private function, see line 1441.

```typescript
const details = await Private.showGitOperationDialog<IGitCloneArgs>(
```

Is it ok to export this function?

Reply:
I did not move it out because there was a reference to a private function, see line 1441.

```typescript
const details = await Private.showGitOperationDialog<IGitCloneArgs>(
```

Is it ok to export this function?

Reply:
Yes please remove the namespace `Private` and export the `showGitOperationDialog` function directly.

Reply:
Done!
Would you be able to cut a release when after you merge?

Reply:
Yes, I have don't the changes I wanted on master. So after merging I only need to bump the version, test and publish.

Reply:
Thanks a lot for helping with this.